### ISSUE TITLE
Links im Footer müssen auch ohne Login funktionieren

### ISSUE THREAD

**Issue by fbufbi** [bkd-mba-fbi/evento-portal#190]
Die Links im Footer müssen auch ohne Login funktionieren, damit im OAuth Loginform darauf verwiesen werden kann (Findings aus #167). 

- [ ]  module=contact
- [ ] module=legal
- [ ] module=imprint

**Comment by caebr** [comment]
@fbufbi 
Das aktuelle Layout dieser Seiten enthält heute Informationen, welche ohne Login nicht verfürbar sind -  eigentlich alles ausser dem Logo und dem eigentlichen Content.
Technisch wäre es sehr aufwändig, dies ohne Login zu ermöglichen.

Wir sind uns noch nicht sicher, was technisch die beste Lösung ist. Diese muss im Rahmen der Umsetzung noch erarbeitet werden.

Varianten:
- Die drei oben erwähnten Inhaltsmodule bzw. Web Components auf der Login-Seite einbinden (inkl. Routing)
- Separates index.html mit Layoutkomponente, welches ohne Authentifizierung funktionert (unter anderer URL verfügbar)
- ...

**Comment by fbufbi** [comment]
@schefbi Was meinst du zum Kommentar? Falls technisch zu aufwändig, müssen wir dann doch bei BKD hidden websites verlangen.

**Comment by schefbi** [comment]
> @schefbi Was meinst du zum Kommentar? Falls technisch zu aufwändig, müssen wir dann doch bei BKD hidden websites verlangen.

Mögliche Lösung: Wir stellen die Seiten auf dem OAuth Server zur Verfügung und linke die Footer Seiten im Portal auf die OAuth Server Seiten. 

Nachteil: Wir müssen das im Custom teil des OAuth Server machen unsere Regel ist eigentlich soviel wie nötig und so wenig wie möglich im Custom teil anpassen.

**Comment by hupf** [comment]
@fbufbi @schefbi Vielleicht noch etwas konkreter, was wir sagen wollten ist:
- Wir würden nicht das bestehende Evento Portal mit dem Authentifizierungsmechanimus so anpassen, damit diese Inhaltsseiten ohne Login verfügbar werden. Dies weil der Aufwand dafür zu gross wäre und dies eine grosse Komplexität (und Fehlerquelle) einführen würde.
- Was wir als realistisch sehen ist ein neuer Einstiegspunkt (eigene Layoutkomponente welche zu einem grossen Teil die vorhandenen Header, Footer etc. Komponent verwenden kann) zu erstellen, ohne Authentifizierungsmechanismus. Dies wäre allerdings nicht in einem halben Tag gemacht, sondern schätzungsweise eher in 2-4PT.

Die Frage bei einer Variante wo diese Content Seiten ausserhalb des Portals erstellt werden wäre:
- Belassen wir sie zusätzlich im Portal so wie sie aktuell sind und pflegen sie doppelt? → einfach umzusetzen
- Verlinken wir im Portal direkt darauf? Was passiert dann allerdings wenn man eingeloggt ist mit der Naviation etc.? → scheint keine ideale Lösung zu geben

**Comment by fbufbi** [comment]
@hupf Danke für die Präzisierung. Die 2-4 PT sind uns zu viel. Wir stellen die Seiten zusätzlich im OAuth Server zur Verfügung und verlinken von der Login-Maske dorthin. D.h. der Task ist bei uns und fällt für euch weg,

**Comment by fbufbi** [comment]
> @fbufbi Ist es richtig, dass wir die Verlinkung auf der Login Seite steuern können? Wir könnten dort also noch einen weiteren URL Parameter hinzufügen oder eine anderen Pfad verwenden als für die eingeloggte Seite?

@hupf Ich würde meinen ja. Der Footer ist zusätzlich in der Loginform umgesetzt und deshalb nicht die gleiche Komponente wie wenn eingeloggt. @schefbi müsste dann wohl die Verlinkung nach eurem Vorschlag in der Loginform anpassen.

**Comment by hupf** [comment]
@fbufbi Ich habe heute einen POC gemacht und ich habe das Gefühl eine Lösung gefunden zu haben die in vernüftiger Zeit umsetzbar ist.

**Comment by fbufbi** [comment]
@schefbi Musst du nun noch im Login-Form die Links austauschen?

**Comment by hupf** [comment]
> @schefbi Musst du nun noch im Login-Form die Links austauschen?

Ja, die Links müssen auf `/unrestricted/?locale=...&module=...` angepasst werden.

**Comment by schefbi** [comment]
Test OK. Links im custom OAuth Template angepasst.

**Comment by schefbi** [comment]
@hupf : Auf Netlify funktioniert das weiterhin nicht.

https://evt-test.apps.be.ch/unrestricted/?locale=de-CH&module=contact

**Comment by hupf** [comment]
@schefbi Ich habe das Docker Image welches von der Pipeline gebaut wurde (https://github.com/bkd-mba-fbi/evento-portal/pkgs/container/evento-portal/338691764?tag=1.30.0) lokal gepullt und gestartet. Ich sehe, dass im Container die Files am richtigen Ort liegen und beobachte wie das korrekte index.html ausgeliefert wird wenn ich `/unrestricted` requeste. Die Inhaltsseiten funktionieren also lokal mit diesem Image einwandfrei.

Kann es sein, dass in den beiden Setups (Netlify & Bedag) ein Proxy im Spiel ist, der jeweils immer `/index.html` ausliefert, statt `/unrestricted/index.html`?

**Comment by fbufbi** [comment]
@schefbi Wie gehen wir hier weiter? Verschieben nach Tscheppa?

**Comment by schefbi** [comment]
@hupf Ja auf Netlify ist das File vorhanden. Ich werde einmal schauen ob ich noch etwas einstellen kann.

<img width="1082" alt="Image" src="https://github.com/user-attachments/assets/3874c0ae-3882-4ffb-b534-3cb80dd129a5" />

**Comment by fbufbi** [comment]
@hupf Folgendes Verhalten:
1. Gehe zu https://eventoapp-test.apps.be.ch/ (Anmelden Evento Web TEST)
2. Klicke auf einen Link im Footer, z.B. Kontakt (nichts passiert)
3. Anmelden mit l1
4. Resultat: URL lautet fälschlicherweise auf `https://evt-test.apps.be.ch/unrestricted/?locale=de-CH&module=contact#/ `(unrestricted) und Logo kann nicht gefunden werden
![Image](https://github.com/user-attachments/assets/daa91294-3247-44ed-9a97-021431e753f0)

**Comment by hupf** [comment]
> [@hupf](https://github.com/hupf) Folgendes Verhalten:
> 
>     1. Gehe zu https://eventoapp-test.apps.be.ch/ (Anmelden Evento Web TEST)
> 
>     2. Klicke auf einen Link im Footer, z.B. Kontakt (nichts passiert)
> 
>     3. Anmelden mit l1
> 
>     4. Resultat: URL lautet fälschlicherweise auf `https://evt-test.apps.be.ch/unrestricted/?locale=de-CH&module=contact#/ `(unrestricted) und Logo kann nicht gefunden werden
>        ![Image](https://github.com/user-attachments/assets/daa91294-3247-44ed-9a97-021431e753f0)

@fbufbi Genau, das hat mit dem Problem zu tun, dass aus mit nicht bekannten Gründen, in diesen Umgebungen statt `/unrestricted/index.html` das File `/index.html` ausgeliefert wird, deshalb das Redirect. Dies kann ich lokal mit dem Docker-Image, welches identisch deployed worden sein sollte, nicht nachspielen. Nach mir sieht vom Image und von den Files her alles gut aus, ich denke es muss ein Problem bei diesen Umgebungen sein (Proxy, Firewall oder irgend so etwas). @schefbi hat im letzten Kommentar geschrieben, dass er es am anschauen ist bei Netlify und Bedag.

**Comment by hupf** [comment]
@fbufbi @schefbi Moment, ich habe noch eine Idee. Es könnte mit dem Service Worker im Zusammenhang sein. Ich schaue es noch einmal an...

**Comment by hupf** [comment]
Es hatte tatsächlich mit dem Service Worker zu tun, sorry. Habe jetzt `/unrestricted/` davon ausgeschlossen. Lokal musste ich noch einmal einloggen und wieder ausloggen, damit es funktioniert hat. Es muss zuerst der neue Service Worker installiert werden können. Aber ich denke in der Praxis sollte das kein Problem sein, da ja der Stand mit den Footer-Link noch nicht produktiv deployed wurde.

**Comment by hupf** [comment]
@fbufbi @schefbi Der Fix ist gemerged. Und sorry noch einmal, ich brauchte einige Zeit, bis ich realisierte, dass der Service Worker hier sein Unwesen treibt...

**Comment by schefbi** [comment]
@hupf Merci, netlify und Test Umgebung funktionieren nun einwandfrei.
